### ISSUE TITLE
Error resulting in illegal fen castling rights in 1.4.1

### ISSUE THREAD

**Issue by Mercutio1243** [alexobviously/bishop#77]
I have encountered an error in fen castling rights for Standard Chess Variant:

When starting from 
> fen rnbqk1nr/ppp1ppb1/6p1/3p4/8/2N5/PPPPPPP1/R1BQKBNR w KQkq - 0 5

and making the move 

> Rxh8

the resulting fen 

>rnbqk1nR/ppp1ppb1/6p1/3p4/8/2N5/PPPPPPP1/R1BQKBN1 b Qkq - 0 5

is illegal, as it proposes castling rights for black on the kingside (small k in Qkq). However, there is no black rook to castle with on the king side. Maybe the algorithm is confused when a rook is captured by an enemy rook?

**Code example:**
```
bishop.Game bishopGameAnalysisInstance = bishop.Game(fen: "rnbqk1nr/ppp1ppb1/6p1/3p4/8/2N5/PPPPPPP1/R1BQKBNR w KQkq - 0 5");
bishopGameAnalysisInstance.makeMoveSan("Rxh8");
print("Incorrect fen: ${bishopGameAnalysisInstance.fen}");

-> Output: Incorrect fen: rnbqk1nR/ppp1ppb1/6p1/3p4/8/2N5/PPPPPPP1/R1BQKBN1 b Qkq - 0 5
```

If you have time to have a look into this, I would be super grateful. There is no good way to handle this from the outside, and it results in random behavior and crashes (for instance Stockfish engine crashes sometimes but not always when receiving these fens).

**Comment by Mercutio1243** [comment]
I have set up an validator method to check whether illegal castling rights are present. Not very neat or efficient but works for me. So, no stress with fixing this. :) 

But, once you know what you are looking for, you will find more: Below, some other positions, where the error occurs:

> R3kb1r/1b3pp1/1q2p3/1ppnN3/3P4/1BP3Pp/1P2QP1P/2B2RK1 b kq - 0 19
> R3k2r/2nqppb1/1p5p/2p1Pbp1/8/1N1P1N1P/1PP2PP1/2BQK2R b Kkq - 0 15
> rn2k2R/5pb1/4p1p1/qpPpP3/p1n2P2/P1NB1Q2/1PP3P1/R1B1K3 b Qkq - 0 18
> R2qkb1r/1b3ppp/2n1p3/1pp1Pn2/3pB3/2P2N2/1P1P1PPP/1NBQ1RK1 b kq - 0 12
> R3k2r/1bq1bp1p/2n1p1pQ/1pp5/4B3/3P4/1P3PPP/1NB1R1K1 b kq - 0 18

The hypothesis gets stronger that the error is related to rooks capturing enemy rooks for which castling rights were present or possibly just standing on the relevant square.

**Comment by Mercutio1243** [comment]
Et voila, I can confirm that the error occurs for both players both on kingside and queenside when a rook captures enemy rook while the corresponding castling rights still present:

> rnbqkbnR/pppppp2/8/6P1/6p1/8/PPPPPP2/RNBQKBN1 b Qkq - 0 4
> rnbqkbn1/pppppp2/8/6P1/6p1/5P2/PPPPP3/RNBQKBNr w KQq - 0 5
> Rnbqkbnr/2pppppp/8/1P6/1p6/8/2PPPPPP/1NBQKBNR b Kkq - 0 4
> 1nbqkbnr/2pppppp/8/1P6/1p6/2P5/3PPPPP/rNBQKBNR w KQk - 0 5

I explicitly made each rook capture move and immediately bishop returns the fen above with illegal castling rights.

The error does not occur when capturing with a bishop or queen.

**Comment by alexobviously** [comment]
Fixed it - will be in `1.4.2`!
